api/src/main/java/com/epam/pipeline/manager/preprocessing/NgsPreprocessingManager.java

```diff
@@ -0,0 +1,347 @@
+                new PipeConfValue(String.format(EntityTypeField.ARRAY_TYPE, sampleMetadataClass),
+                        JsonMapper.convertDataToJsonStringForQuery(
+                                samples.stream().map(MetadataEntityVO::getExternalId).collect(Collectors.toList())
+                        )
+                ));
+        final MetadataParsingResult toUpd = new MetadataParsingResult(
+                machineRunMetadataEntity.getClassEntity(),
+                Collections.singletonMap(sampleMetadataClass,
+                        samples.stream().map(MetadataEntityVO::getExternalId).collect(Collectors.toSet())
+                ),
+                Collections.singletonMap(machineRunMetadataEntity.getExternalId(), machineRunMetadataEntity)
+        );
+        metadataEntityManager.createAndUpdateEntities(folderId, toUpd);
+    }
+
+    private MetadataEntity fetchMachineRunMetadataEntity(final DataStorageLink dataFolderPath,
+                                                         final Long machineRunId) {
+        final String machineRunMetadataClass = preferenceManager.getStringPreference(
+                SystemPreferences.PREPROCESSING_MACHINE_RUN_CLASS.getKey());
+
+        final MetadataEntity machineRunMetadataEntity = metadataEntityManager.load(machineRunId);
+        Assert.notNull(machineRunMetadataEntity,
+                messageHelper.getMessage(
+                        MessageConstants.ERROR_NGS_PREPROCESSING_NO_MACHINE_RUN_METADATA, machineRunId));
+        Assert.state(machineRunMetadataEntity.getClassEntity().getName().equals(machineRunMetadataClass),
+                messageHelper.getMessage(MessageConstants.ERROR_NGS_PREPROCESSING_MACHINE_RUN_WRONG_METADATA_CLASS,
+                machineRunId, machineRunMetadataEntity.getClassEntity().getName(), machineRunMetadataClass));
+
+        // check that folder for machineRun exists
+        if (!checkPathExistence(dataFolderPath.getDataStorageId(),
+                Paths.get(dataFolderPath.getPath(), machineRunMetadataEntity.getExternalId()).toString())) {
+            throw new IllegalStateException(
+                    messageHelper.getMessage(MessageConstants.ERROR_DATASTORAGE_PATH_NOT_FOUND,
+                            dataFolderPath.getPath(), dataFolderPath.getDataStorageId()));
+        }
+        return machineRunMetadataEntity;
+    }
+
+    private DataStorageLink fetchDataFolder(final MetadataEntry metadata) {
+        final String dataFolderMetadataKey = preferenceManager.getStringPreference(
+                SystemPreferences.PREPROCESSING_DATA_FOLDER.getKey());
+        final PipeConfValue dataPath = metadata.getData().get(dataFolderMetadataKey);
+        Assert.notNull(dataPath,
+                messageHelper.getMessage(MessageConstants.ERROR_NGS_PREPROCESSING_FOLDER_SHOULD_HAVE_METADATA,
+                        dataFolderMetadataKey)
+        );
+        storageManager.analyzePaths(Collections.singletonList(dataPath));
```

Comment:
@mzueva I have added second one realization of `analyzePaths` which doesn't load all sorages but tries to load storage by prefix from `PipeConfValue`. I decide to do in that why because using here `DataStorageLink` is super convenient. What do you think?